harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -49,8 +67,8 @@ class Checkpoint(object):
     def __init__(
         self,
         session: api.Session,
-        task_id: str,
-        allocation_id: str,
+        task_id: Optional[str],
+        allocation_id: Optional[str],
```

Comment:
Are these conceptually (semantically) optional?

Reply:
proto just has no way to express that this entire struct is either None, or all the fields are present.

But that's conceptually what the system guarantees (either the TrainingMetadata isn't there or it's fully populated), so we promise that here in the sdk.

Reply:
Oh!  I was talking about something else when I answered.

task_id and allocation_id are optional because we want to support importing checkpoints into the system which are not related to any task or allocation.